app/src/components/UI/Molecules/MultiSelectFilter/CuisineFilter.tsx

```diff
@@ -0,0 +1,104 @@
+import React from "react";
+import { Dropdown } from "semantic-ui-react";
+import { useAppDispatch, useAppSelector } from "../../../../store/root";
+import {
+  getUserIDFromToken,
+  useCreateRecommendationMutation,
+} from "../../../../api";
+import { setCuisineType } from "../../../../store/search";
+import styles from "./filter.module.css";
+import { v4 as uuid } from "uuid";
+
+const options = [
+  {
+    key: "Mexican",
+    text: "Mexican",
+    value: "mexican",
+  },
+  {
+    key: "Indian",
+    text: "Indian",
+    value: "indian",
+  },
+  {
+    key: "Chinese",
+    text: "Chinese",
+    value: "chinese",
+  },
+  {
+    key: "Japanese",
+    text: "Japanese",
+    value: "japanese",
+  },
+  {
+    key: "French",
+    text: "French",
+    value: "french",
+  },
+  {
+    key: "Spanish",
+    text: "Spanish",
+    value: "spanish",
+  },
+  {
+    key: "Thai",
+    text: "Thai",
+    value: "thai",
+  },
+  {
+    key: "Korean",
+    text: "Korean",
+    value: "korean",
+  },
+  {
+    key: "Italian",
+    text: "Italian",
+    value: "italian",
+  },
+];
+
+interface Props {
+  searchQuery?: string;
+}
+
+const CuisineFilter: React.FC<Props> = ({ searchQuery }) => {
+  const dispatch = useAppDispatch();
+  const [createRecommendation] = useCreateRecommendationMutation();
+  const token = useAppSelector((state) => state.token.token);
+
+  let userID = null as string | null;
+  if (token) {
+    userID = getUserIDFromToken(token);
+  }
+
+  return (
+    <Dropdown
+      placeholder="Cuisine Type"
+      fluid
+      multiple
+      search
+      selection
+      options={options}
+      className={styles.filter}
+      onChange={(e, data) => {
+        dispatch(setCuisineType(data.value as string[]));
+        if (token === null) {
+          throw new Error("token is null");
+        }
+        const userID = getUserIDFromToken(token);
```

Comment:
I'm not sure why you need to get `userID` a second time here since you got it already in line 71.
The way you call `createRecommendation()` here looks like a bad anti-pattern. Why not call it in the same place we are calling the `createQuery` mutation?

Reply:
Which file is createQuery being called?

Reply:
Submission handler for SearchBox.

Reply:
Actually, `createQuery` only depends on the search string while `createRecommendation` depends on the filters as well. That makes sense why it is used in `CuisineFilter`.

Reply:
I worry that it would result in a mass amount of `PastSearch` records being created if it's created for every time the filter is changed.

Reply:
I have the createRecommendation being called in cuisinefilter, because I want the recommendation to be generated immediately after the cuisine type is changed.